### PR TITLE
Add support for specifying a seed for the monkey command

### PR DESCRIPTION
**Pull request by emanuelez**
This PR allows to specify the seed to run the monkey command with.
This allows to be able to reproduce failing monkey runs for debugging purposes.

### CODE REVIEW

File: gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/AndroidCommandPlugin.groovy [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class AndroidCommandPlugin implements Plugin<Project> {
         def extension = project.android.extensions.create("command", AndroidCommandPluginExtension, project)
         extension.task 'installDevice', 'installs the APK on the specified device', Install, ['assemble']
         extension.task 'run', 'installs and runs a APK on the specified device', Run, ['installDevice']
-        extension.task 'monkey', 'calls the monkeyrunner on the specified device', Monkey, ['installDevice']
+        extension.task 'monkey', 'calls the monkey command on the specified device', Monkey, ['installDevice']
```
Comment (devisnik) [inline gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/AndroidCommandPlugin.groovy]: :+1:

File: gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/Monkey.groovy [hunk 2]
```diff
@@ -22,4 +26,10 @@ class Monkey extends AdbTask {
             events = events.call()
         events ?: pluginEx.events
     }
+
+    private getSeed() {
+        if (seed instanceof Closure)
+            seed = seed.call()
+        seed ?: pluginEx.getSeed()
```
Comment (devisnik) [inline gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/Monkey.groovy]: Might use `pluginEx.seed` as this is the standard property access. It results in the getter being called.

File: gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy [hunk 3]
```diff
@@ -14,7 +14,17 @@ class AndroidCommandPluginExtensionTest extends GroovyTestCase {
...
 
-    private AndroidCommandPluginExtension createExtension() {
+    void testDefaultEvents() {
+        def extension = createExtension()
+        assert extension.getEvents() == 10000
+    }
+
+    void testDefaultSeed() {
+        def extension = createExtension()
+        assert extension.getSeed() == null
```
Comment (devisnik) [inline gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy]: Or `assert !extension.getSeed()` ?

File: gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy [hunk 4]
```diff
@@ -14,7 +14,17 @@ class AndroidCommandPluginExtensionTest extends GroovyTestCase {
...
+    }
+
+    private static AndroidCommandPluginExtension createExtension() {
```
Comment (devisnik) [inline gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy]: :+1: for `static`

**Comment by devisnik** [comment]
Nice! 
Could you please add some basic PR description?

File: gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy (hunk 3 above)
Comment (emanuelez) [inline gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy]: Yeah... I just wanted to make it very explicit that we expect `null` and not `false` or anything else.

Comment (devisnik) [inline gradle-android-command-plugin/src/test/groovy/com/novoda/gradle/command/AndroidCommandPluginExtensionTest.groovy]: Fine then.

**Comment by stefanhoth** [comment]
That looks good! Thank you for this contribution.

Our CI projects will soon have something like this thanks to you:

```
./gradlew clean monkeyDebug -Dseed=$BUILD_NUMBER
```

**Comment by emanuelez** [comment]
Yep... that's exactly what I had in mind :)

**Pull request merged**